Forge.TreeWalker/src/TreeWalkerSession.cs

```diff
@@ -406,43 +438,66 @@ public async Task<string> WalkTree(string treeNodeKey)
         /// Visits a TreeNode in the ForgeTree, performing type-specific behavior as necessary before selecting the next child to visit.
         /// </summary>
         /// <param name="treeNodeKey">The TreeNode key to visit.</param>
+        /// <param name="skipAction">Specifies whether to skip all actions and do ChildSelector only.</param>
         /// <exception cref="TimeoutException">If the node-level timeout was hit.</exception>
         /// <exception cref="ActionTimeoutException">If the action-level timeout was hit.</exception>
         /// <exception cref="OperationCanceledException">If the cancellation token was triggered.</exception>
         /// <exception cref="Exception">If an unexpected exception was thrown.</exception>
         /// <returns>The key of the next child to visit, or <c>null</c> if no match was found.</returns>
         public async Task<string> VisitNode(string treeNodeKey)
+        {
+            return await this.VisitNode(treeNodeKey, skipAction: false);
```

Comment:
good tips, and it can save one more context switch by saving one async await.

---
In reply to: [504169506](https://github.com/microsoft/Forge/pull/32#discussion_r504169506) [](ancestors = 504169506)